Codeinwp/otter-blocks
# Form UI/UX Enhancement

Closes #1330.


### Summary


New UI and UX. Better integration with Global Defaults.

Explication for the codes that appears on the front for clients:

- 101: The request is missing data like Form Id/Option, page URL, or nonce value - this can be missing when the Nonce Field in the form was manually deleted with force. Also, the given nonce might be invalid.
- 102: No given captcha token when Bot Protection is set to active.
- 104: The email field is missing. This is requested when the Form is set to subscribe the user to a mail list
- 106: The email could not be sent. Might be a problem with the mail service.
- 108: The API key for the Marketing Integration service is missing. Either it was deleted by hand or a script.
- 201: An external script has changed the settings, and the accessed service is no longer present.
- 202: Error from the service when trying to subscribe the user, like servers are unavailable, rejected request, etc.
- 203: The 3rd-party authentication is invalid. Might be a problem with API Key: expired, rejected, etc. Need to check with the provider.
- 204: The provided captcha token is invalid: the user might be a bot.
- 205: The API Key format is invalid. The client might not have copied the right key, or the provider changed the format, and we did not update the check.
- 206: The email was already registered in the Marketing Integration service.
- 207: The service rejected the given email by the user. It might consider it fake.

### Screenshots 

----

### Test instructions


This is a UI refresh; we need to ensure that all style options are usable.

1. Make sure that the old blocks don't break.
2. Test the Sidebar and the responsive options.
3. Test the Global defaults

With this update, we let the user set a message in case of an error. The reason for the error will be displayed in the Console along with its code. Like this:

![image](https://user-images.githubusercontent.com/17597852/210768058-e7496734-94ff-4fe1-833e-8ddfefb687ed.png)

The basic email functionality should be effaced by this. If you happen to have a problem, please put an SS with an error in the console. The error code is vital for debugging.



---- 

### Checklist before the final review
